src/picongpu/include/particles/bremsstrahlung/Bremsstrahlung.hpp

```diff
@@ -0,0 +1,173 @@
+
+    /* relevant area of a block */
+    typedef SuperCellDescription<
+        typename MappingDesc::SuperCellSize,
+        LowerMargin,
+        UpperMargin
+        > BlockArea;
+
+    BlockArea BlockDescription;
+
+    typedef MappingDesc::SuperCellSize TVec;
+
+    typedef FieldTmp::ValueType ValueTypeIonCounter;
+
+private:
+    /* global memory ion density field device databoxes */
+    PMACC_ALIGN(ionCounterBox, FieldTmp::DataBoxType);
+    /* shared memory EM-field device databoxes */
+    PMACC_ALIGN(cachedIonCounter, DataBox<SharedBox<ValueTypeIonCounter, typename BlockArea::FullSuperCellSize, 0> >);
+
+    PMACC_ALIGN(scaledSpectrumFunctor, ScaledSpectrum::LookupTableFunctor);
+    PMACC_ALIGN(stoppingPowerFunctor, ScaledSpectrum::LookupTableFunctor);
+    PMACC_ALIGN(getPhotonAngleFunctor, GetPhotonAngle::GetPhotonAngleFunctor);
+
+    PMACC_ALIGN(photonMom, float3_X);
+    PMACC_ALIGN(softPhotonsCutoff, float_X);
+
+    /* random number generator */
+    typedef PMacc::random::RNGProvider<simDim, PMacc::random::methods::XorMin> RNGFactory;
+    typedef PMacc::random::distributions::Uniform<float> Distribution;
+    typedef typename RNGFactory::GetRandomType<Distribution>::type RandomGen;
+    RandomGen randomGen;
+
+public:
+    /* host constructor initializing member : random number generator */
+    Bremsstrahlung(
+        const ScaledSpectrum::LookupTableFunctor& scaledSpectrumFunctor,
+        const ScaledSpectrum::LookupTableFunctor& stoppingPowerFunctor,
+        const GetPhotonAngle::GetPhotonAngleFunctor& getPhotonAngleFunctor,
+        const uint32_t currentStep);
+
+    /** Initialization function on device
+     *
+     * \brief Cache ion density field on device
+     *         and initialize possible prerequisites, like e.g. random number generator.
+     *
+     * This function will be called inline on the device which must happen BEFORE threads diverge
+     * during loop execution. The reason for this is the `__syncthreads()` call which is necessary after
+     * initializing the E-/B-field shared boxes in shared memory.
+     */
+    DINLINE void init(const DataSpace<simDim>& blockCell, const int& linearThreadIdx, const DataSpace<simDim>& localCellOffset);
+
+    /** Rotates a vector to a given polar angle and a random azimuthal angle.
+     *
+     * @param vec vector to be rotated
+     * @param theta polar angle
+     * @return rotated vector
+     */
+    DINLINE float3_X scatterByTheta(const float3_X vec, const float_X theta);
+
+    /** Return the number of target particles to be created from each source particle.
+     *
+     * Called for each frame of the source species.
+     *
+     * @param sourceFrame Frame of the source species
+     * @param localIdx Index of the source particle within frame
+     * @return number of particle to be created from each source particle
+     */
+    DINLINE unsigned int numNewParticles(FrameType& sourceFrame, int localIdx);
```

Comment:
@Heikman Could your target particles be something other than photons? If not - why not call this more clearly `numNewPhotons`?